src/scmcoat/core.py

```diff
@@ -193,7 +193,37 @@ def _run(self, emiss, useMultigas=True):
                     -1, :
                 ]  # hold the last element constant for the rest of the array
             else:
-                raise NotImplementedError("Unable to handle emissions other than lengths 736 or 751")
+                nt = emiss.shape[0] # assume time length is same as input emissions. assume no change in ref year
+                argslength = args["F_solar"].shape[0] # assume solar, volc, natural have same time dim
+                
+                # assuming solar/volc/nat have same start year as emiss
+                if argslength < nt:
```

Comment:
This logic is great. However, I think some logic needs to be added to handle different start years. Otherwise, natural emissions would be put into the wrong time slots, and there might be other issues.